test/e2e/aws_test.go

```diff
@@ -118,6 +140,172 @@ var _ = Describe("functional tests", func() {
 	})
 })
 
+func scaleMachineDeployment(namespace, machineDeployment string, replicasCurrent int32, replicasProposed int32) {
+	fmt.Fprintf(GinkgoWriter, "Scaling MachineDeployment from %d to %d\n", replicasCurrent, replicasProposed)
+	deployment := &clusterv1.MachineDeployment{}
+	Expect(kindClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: namespace, Name: machineDeployment}, deployment))
+	deployment.Spec.Replicas = &replicasProposed
+	Expect(kindClient.Update(context.TODO(), deployment)).NotTo(HaveOccurred())
+	waitForMachinesCountMatch(namespace, machineDeployment, replicasCurrent, replicasProposed)
+	waitForMachineDeploymentRunning(namespace, machineDeployment, replicasProposed)
+}
+
+func createMachineTemplate(namespace, clusterName, machineDeploymentName, awsMachineTemplateName, bootstrapConfigName string,nodepool string, replicas int32) {
+	fmt.Fprintf(GinkgoWriter, "Creating MachineDeployment in namespace %s with name %s\n", namespace, machineDeploymentName)
+	makeAWSMachineTemplate(namespace, awsMachineTemplateName)
+	makeJoinBootstrapConfigTemplate(namespace, bootstrapConfigName)
+	makeMachineDeployment(namespace, machineDeploymentName, awsMachineTemplateName, bootstrapConfigName, clusterName, nodepool, replicas)
+	waitForMachinesCountMatch(namespace, machineDeploymentName,replicas, replicas)
+	waitForMachineDeploymentRunning(namespace, machineDeploymentName, replicas)
+}
+
+func waitForMachinesCountMatch(namespace, machineDeploymentName string, replicasCurrent int32, replicasProposed int32) {
+	fmt.Fprintf(GinkgoWriter, "Ensuring Machine count matched to %d \n", replicasProposed)
+	machineDeployment := &clusterv1.MachineDeployment{}
+	if err := kindClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: namespace, Name: machineDeploymentName}, machineDeployment); err != nil {
+		return
+	}
+	scalingDown := replicasCurrent > replicasProposed
+	Eventually(
+		func() (int32, error){
+			machineList := &clusterv1.MachineList{}
+			selector, err := metav1.LabelSelectorAsMap(&machineDeployment.Spec.Selector)
+			if err!=nil {
+				return -1,err
+			}
+			if err := kindClient.List(context.TODO(), machineList, crclient.InNamespace(namespace), crclient.MatchingLabels(selector)); err != nil {
+				return -1,err
+			} else {
+				var runningMachines int32
+				var deletingMachines int32
+				for _,item := range machineList.Items {
+					if string(clusterv1.MachinePhaseRunning) == item.Status.Phase {
+						runningMachines += 1
+					}
+					if string(clusterv1.MachinePhaseDeleting) == item.Status.Phase {
+						deletingMachines += 1
+					}
+				}
+				if scalingDown {
+					if deletingMachines > 0{
+						return -1, nil
+					}
+				}
+				return runningMachines, nil
+			}
+
+		},
+		20*time.Minute, 30*time.Second,
+	).Should(Equal(replicasProposed))
+}
+
+func waitForMachineDeploymentRunning(namespace, machineDeploymentName string, replicas int32) {
```

Comment:
This can probably just test to make sure that `machineDeployment.Spec.Replicas` == `machineDeployment.Status.ReadyReplicas` instead of needing to pass in `replicas`

Reply:
Did the change to compare Replicas with ReadyReplicas and removed replicas method parameter